evm/db/journal.py

```diff
@@ -130,42 +164,34 @@ class JournalDB(BaseDB):
     wrapped_db = None
     journal = None
 
-    def __init__(self, wrapped_db):
+    def __init__(self, wrapped_db: BaseDB) -> None:
         self.wrapped_db = wrapped_db
         self.journal = Journal()
+        # Not sure if that's the right thing to do but for now assume
+        # we just start recording right away.
+        self.record()
 
-    def get(self, key):
-        return self.wrapped_db.get(key)
+    #TODO: Discuss potential inefficiency issue
+    def get(self, key: bytes) -> bytes:
```

Comment:
@pipermerriam unrelated to the bigger picture but as a little side discussion. I *think* not finding a key in the journal and further having to look it up in the underlying database won't be an exceptional case but rather be something quite normal. Lookups in the journal a quite costly in itself because we have to loop through the checkpoints so I wonder if throwing a `KeyError` and then catching it to lookup the underlying db maybe inefficient?

Reply:
Not sure, try it out?

Reply:
> Lookups in the journal a quite costly in itself because we have to loop through the checkpoints

I think this cost is orders of magnitude lower than querying the underlying leveldb.

Reply:
>I think this cost is orders of magnitude lower than querying the underlying leveldb.

Definitely faster than that. But I *think* the lookups could be even faster without the exception ping pong (because stack traces etc) But again, that's really just a side discussion and I can measure it once I got the basics right.

Reply:
Ah, I mis-understood.  Yes, this performance gain is desirable.